ci: add ci config

TGiles opened the pull request:
This patch adds the [Circle CI configuration](https://github.com/mozilla-services/Dockerflow/blob/03e880ec476d9f8b5fc21378b93fb039bab89316/.circleci/config.yml) and the [Dockerfile](https://github.com/mozilla-services/Dockerflow/blob/03e880ec476d9f8b5fc21378b93fb039bab89316/Dockerfile) from the [mozilla-services/dockerflow repository](https://github.com/mozilla-services/Dockerflow).

@smarnach I'm not entirely sure how to verify that the Circle CI configuration and the Dockerfile are valid for this (i.e. not sure how to test my changes), do you have some advice for testing locally and ensuring these changes are valid?

smarnach commented on Dockerfile:
```diff
@@ -0,0 +1,21 @@
+    rm -rf ~app/.node-gyp
+
+# Finally copy in the app's source file
+COPY . /app
```
You probably only need to copy `update-script.js`, right?

TGiles replied:
Yeah that sounds about right, thanks for catching that!

TGiles commented:
Hey Sven, this should be ready for another review. Fixed the copying issue and updated the Node version on the Docker image, since my script needs Node 14.

smarnach commented:
@TGiles Looks great! I'll need to set up Circle CI and Docker Hub next, but I'll probably only get to that next week. I filed https://jira.mozilla.com/browse/SE-1735 for this work.

TGiles commented:
@smarnach Thanks for feedback, and thanks for filing that JIRA work! Next week is more than okay for me, not a high priority to say the least

smarnach reviewed (approved):
I took the liberty to fix remove the test job from the Circle CI config – there are no unit tests, and `npm test` fails, causing the build to fail.

To test whether pushing images to Docker Hub works fine, I need to merge this PR.

smarnach commented:
Everything worked fine: https://hub.docker.com/layers/mozilla/passwordmgr-related-realms-updater/latest/images/sha256-6a0ed6db404e7d348c80ce3af197e65376f9247d8f93a859e1e91a00687c334b?context=repo

smarnach commented:
I noticed that I'm out for all of next week, and I unexpectedly had some free time today, so I finished this earlier than planned. The next step would be to set up a Jenkins job to run this Docker image.
